File: Soal1.php
<!DOCTYPE html>
<html>
<head>
    <title>Inheritance Bangun Datar</title>
</head>
<body>
    <form method="post" action="">
        <label for="jenis_bangun">Pilih jenis bangun datar:</label>
        <select name="jenis_bangun" id="jenis_bangun">
            <option value="persegi">Persegi</option>
            <option value="persegipanjang">Persegi Panjang</option>
            <option value="segitiga">Segitiga</option>
            <option value="lingkaran">Lingkaran</option>
        </select>
        <input type="submit" name="submit" value="Submit">
    </form>
    <?php
    if (isset($_POST['submit'])) {
        $jenis_bangun = $_POST['jenis_bangun'];
        switch ($jenis_bangun) {
            case 'persegi':
                $sisi = 5; // contoh nilai sisi
                $persegiLuas = $sisi * $sisi;
                $persegiKeliling = 4 * $sisi;
                echo "Luas Persegi: " . $persegiLuas . "<br>";
                echo "Keliling Persegi: " . $persegiKeliling;
                break;
            case 'persegipanjang':
                $panjang = 8; // contoh nilai panjang
                $lebar = 5; // contoh nilai lebar
                $ppLuas = $panjang * $lebar;
                $ppKeliling = 2 * ($panjang + $lebar);
                echo "Luas Persegi Panjang: " . $ppLuas . "<br>";
                echo "Keliling Persegi Panjang: " . $ppKeliling;
                break;
            case 'segitiga':
                $alas = 10; // contoh nilai alas
                $tinggi = 6; // contoh nilai tinggi
                $segitigaLuas = 0.5 * $alas * $tinggi;
                // asumsi segitiga adalah segitiga siku-siku
                $sisiMiring = sqrt($alas * $alas + $tinggi * $tinggi);
                $segitigaKeliling = $alas + $tinggi + $sisiMiring;
                echo "Luas Segitiga: " . $segitigaLuas . "<br>";
                echo "Keliling Segitiga: " . $segitigaKeliling;
                break;
            case 'lingkaran':
                $jariJari = 7; // contoh nilai jari-jari
                $lingkaranLuas = 3.14 * $jariJari * $jariJari;
                $lingkaranKeliling = 2 * 3.14 * $jariJari;
                echo "Luas Lingkaran: " . $lingkaranLuas . "<br>";
                echo "Keliling Lingkaran: " . $lingkaranKeliling;
                break;
            default:
                echo "Pilihan tidak valid.";
                break;
        }
    }
    ?>
</body>
</html>

File: Soal1.cpp
#include <iostream>
#include <cmath>

using namespace std;

class BangunDatar {
public:
    virtual void hitungLuas() = 0;
    virtual void hitungKeliling() = 0;
};

class Persegi : public BangunDatar {
private:
    double sisi;

public:
    Persegi(double s) : sisi(s) {}

    void hitungLuas() {
        cout << "Luas Persegi: " << sisi * sisi << endl;
    }

    void hitungKeliling() {
        cout << "Keliling Persegi: " << 4 * sisi << endl;
    }
};

class PersegiPanjang : public BangunDatar {
private:
    double panjang;
    double lebar;

public:
    PersegiPanjang(double p, double l) : panjang(p), lebar(l) {}

    void hitungLuas() {
        cout << "Luas Persegi Panjang: " << panjang * lebar << endl;
    }

    void hitungKeliling() {
        cout << "Keliling Persegi Panjang: " << 2 * (panjang + lebar) << endl;
    }
};

class Segitiga : public BangunDatar {
private:
    double alas;
    double tinggi;

public:
    Segitiga(double a, double t) : alas(a), tinggi(t) {}

    void hitungLuas() {
        cout << "Luas Segitiga: " << 0.5 * alas * tinggi << endl;
    }

    void hitungKeliling() {
        double sisiMiring = sqrt(alas * alas + tinggi * tinggi);
        cout << "Keliling Segitiga: " << (alas + tinggi + sisiMiring) << endl;
    }
};

class Lingkaran : public BangunDatar {
private:
    double jariJari;

public:
    Lingkaran(double r) : jariJari(r) {}

    void hitungLuas() {
        cout << "Luas Lingkaran: " << 3.14 * jariJari * jariJari << endl;
    }

    void hitungKeliling() {
        cout << "Keliling Lingkaran: " << 2 * 3.14 * jariJari << endl;
    }
};

int main() {
    Persegi persegi(5);
    persegi.hitungLuas();
    persegi.hitungKeliling();

    PersegiPanjang persegipanjang(8, 5);
    persegipanjang.hitungLuas();
    persegipanjang.hitungKeliling();

    Segitiga segitiga(10, 6);
    segitiga.hitungLuas();
    segitiga.hitungKeliling();

    Lingkaran lingkaran(7);
    lingkaran.hitungLuas();
    lingkaran.hitungKeliling();

    return 0;
}
